### PR TITLE
Fixes #32407 - Trim the Metadata json

### PR DESCRIPTION
**Pull request by parthaa**
Trimmed the json to have the products and gpg_keys in separate lists.
Also moved some of the metadata generation to a Metadata Generator
class.

### CODE REVIEW

**Comment by parthaa** [comment]
To test this

- Keep a few products , repos and associated gpg keys
- Run ``` hammer content-export incremental library --organization-id=<org-id> ```
- then cat the generated metadata json. Notice new products/repositories/gpg_keys sections
```
$ cat metadata-23.json | json_reformat
{
    "organization": "export-3439",
    "products": {
       "prod" : { 
                "name": "prod",
                "label": "prod",
                "description": null,
                "gpg_key": {

                }
     },
     "gpg_keys": {
        "great" : : {
                "name": "great",
                "content_type": "gpg_key",
                "content": "great gatsby" 
            },
     }
    "repositories": {
        <.....other mappings .....>
        "misc-33384": {
            "name": "misc",
            "label": "misc",
            "description": null,
            "arch": "x86_64",
            "content_type": "yum",
            "unprotected": true,
            "checksum_type": "sha256",
            "os_versions": [
                "rhel-7" 
            ],
            "major": null,
            "minor": null,
            "product": { "label": "prod" }          
            "gpg_key": { "name": "great" }
            },
            "redhat": false
        }
    },
    "content_view": {
        "name": "Export-Library",
        "label": "Export-Library",
        "description": null
    },
    "content_view_version": {
        "major": 9,
        "minor": 0
    },
    "incremental": false,
    "toc": "export-5e1ca0f7-1af7-49b2-97d5-f9417760d11a-20210421_0159-toc.json" 
}

```

- Run the corresponding import operation and verify it works.

**Comment by parthaa** [comment]
[test katello]

File: app/services/katello/pulp3/content_view_version/export.rb [hunk 1]
```diff
@@ -3,15 +3,16 @@ module Pulp3
     module ContentViewVersion
       class Export
         include ImportExportCommon
+        attr_accessor :smart_proxy, :content_view_version, :destination_server, :from_content_view_version
```
Comment (jturel) [inline app/services/katello/pulp3/content_view_version/export.rb:6]: Sorry if I missed it... Do these need to be accessors ie someone needs to be able to set these values on an instance of this class from the outside?

Comment (jturel) [reply app/services/katello/pulp3/content_view_version/export.rb:6]: Maybe attr_reader would be more appropriate if accessor is not needed

**Comment by jturel** [comment]
Concept looks good! Will test once you comment on my last point / fix tests to avoid churn :)

File: app/services/katello/pulp3/content_view_version/export.rb (hunk 1 above)
Comment (parthaa) [reply app/services/katello/pulp3/content_view_version/export.rb:6]: I could do the reader, that sounds better. I just need them to be accessible to the metadata generator. The generator is using methods from the service class to fetch repository info.

Comment (parthaa) [reply app/services/katello/pulp3/content_view_version/export.rb]: updated

**Comment by parthaa** [comment]
[test katello]

**Review by jturel — APPROVED** [review]
ACK

Works well - much more readable and compact!

**Pull request merged**